src/kubernetes_upgrade.py

```diff
@@ -91,10 +91,13 @@ def _get_unit_healthy_status(
         revision_hash = self._unit_workload_versions[self._unit.name]
         # Example: 6c67d5f56c
         revision_hash = revision_hash.removeprefix(f"{self._app_name}-")
-        message = f'{self._current_versions["charm"]} {revision_hash}'
         if isinstance(workload_status, ops.WaitingStatus):
-            return ops.WaitingStatus(message)
-        return ops.ActiveStatus(message)
+            return ops.WaitingStatus(
+                f'Router {self._current_versions["workload"]}; Charmed operator {self._current_versions["charm"]}; Kubernetes rev {revision_hash}'
+            )
+        return ops.ActiveStatus(
+            f'Router {self._current_versions["workload"]} running; Charmed operator {self._current_versions["charm"]}; Kubernetes rev {revision_hash}'
+        )
```

Comment:
actually, i think waiting status (according to https://juju.is/docs/juju/status: The charm is alive and well, and is waiting for some event to occur. No human intervention required) makes sense. why would k8s be different than vm?

also, i am not sure that showing the kubernetes rev externally to the user is a good idea - we should present them with just the workload version imo (im unsure of even showing the charm version)

Reply:
> if workload is going to upgrade, should unit status be active or waiting?

IMHO: app active, units waiting.
This is what you want to show as SAAS: service is active, but units have upgrade queue.

> why would k8s be different than vm?

I second to have common behavior for VM and K8s.

> Im unsure of even showing the charm version

deploy rev1, refresh rev2 (fail), rollback rev1.

The initial rev1 and the last rev1 will have different K8s revisions => Carl want to show this. It is useful as units rev1/revision_hash1 will have to be restarted to became rev1/revision_hash2.

Q: can we show `Charmed operator 1 (restart pending)` instead of `Charmed operator 1 K8s bcbce12` (which means nothing to users).